# Исследование надежности заемщиков.

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

### Шаг 1. Обзор данных

В двух столбцах есть пропущенные значения. Один из них — days_employed. Пропуски в этом столбце вы обработаете на следующем этапе. Найдите другой столбец и заполните пропущенные значения в нём медианным значением по столбцу:
опишите, какие пропущенные значения вы обнаружили;
проверьте, какую долю составляют пропущенные значения в каждом из столбцов с пропусками;
приведите возможные причины появления пропусков в данных;
объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

1. На первом этапе работы я импортировал библиотеку **pandas** и сократил как **pd** для удобства. 
2. Далее создал переменую **data** в которую я присвоил код с путем к нашей `CSV` таблице.
3. В конце выводим на экран первые 15 строк таблицы

In [1]:
import pandas as pd
data = pd.read_csv('/Users/mmaximmaximovgmail.com/Desktop/DS/Project_2/data.csv')
data.head(15)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   

     total_income                            purpose  
0   253875.639453                      покупка жилья  
1   112080.014102            приобретение автомобиля  
2   145885.952297                      покупка жилья  
3   267628.550329         дополнительное образование  
4   158616.077870                    сыграть свадьбу  
5   255763.565419                      покупка жилья  
6   240525.971920                  операции с жильем  
7   135823.934197                        образование  
8    95856.832424              на проведение свадьбы  
9   144425.938277            покупка жилья для семьи  
10  113943.491460               покупка недвижимости  
11   77069.234271  покупка коммерческой недвижимости  
12            NaN                    сыграть свадьбу  
13  130458.228857            приобретение автомобиля  
14  165127.911772         покупка жилой недвижимости

Получим общую информациб с помощью функции **info()** и **describe()**

In [2]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


**Наблюдение**

По первичному анализу видим, что в столбцах `days_employed` и `total_income` имеются одинаковые отклонения и тип данных

По таблице видно лишние знаки - использован верхний регистр и ошибки в виде `NaN`

In [3]:
data.describe()

children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  1.935100e+04  
mean           0.972544      0.080883  1.674223e+05  
std            1.420324      0.272661  1.029716e+05  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.030532e+05  
50%            0.000000      0.000000  1.450179e+05  
75%            1.000000      0.000000  2.034351e+05  
max            4.000000      1.000000  2.265604e+06

**Наблюдение**

**По столбцу `children`:**
1. Отрицательное значение в `min`, возможно при переносе данных остался минус.
2. Странное максимальное значение в 20 детей, это возможно, но нужно будет проверить как часто данное число втречается и вожно там лишний 0.
**По столбцу `days_employed`:**
1. Первое, что бросается в глаза это знак **-**, это явная ошибка которую нужно в дальнейшем исрпавить для корректного анализа данных.
2. Медианное значение равно 63046 дней или 175 лет, не думаю что у человека может быть такой трудовой стаж, возможно где-то опечатки
**По столбцу `dob_years`:**
Единственное, что бросается в глаза это минимальный возраст клиента и он составляет 0, возможно есть ошибка в данных

Видим, что в двух столбцах, а именно `days_employed` и `total_income`, имеются пропуска и причем по таблице все это находится на одной строке

Теперь посмотрим из разных или одной професии у нас выходят ошибки по столбца `days_employed` и `days_employed`

In [4]:
data[(data['days_employed'].isnull() == True) & (data['total_income'].isnull() == True)]['income_type'].value_counts()

сотрудник          1105
компаньон           508
пенсионер           413
госслужащий         147
предприниматель       1
Name: income_type, dtype: int64

Из разных профессий, значит нужно будет брать медианное значение их всех типов профессий. Теперь посмотрим некоторую информацию по столбцам `children` и `dob_years` 

In [5]:
data[data['dob_years'] == 0].count()[0] #кол-во клиентов возрастом 0 лет


101

In [6]:
data[data['children'] == -1].count()[0] #кол-во людей с -1 ребенокм

47

In [7]:
data[data['children'] == 20].count()[0] #кол-во строк с 20 детьми

76

**Вывод**

1. нулевой возраст клиентов, это банальное не внесение данных в базу, которую нужно исправить
2. -1 ребенок, лишний знак -
3. 20 детей, возможно тут опечатка, хотели написать 2 вместо 20( на исправление)

### Шаг 2.1 Заполнение пропусков

In [8]:
data.isnull().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

Видим, что правда в двух столбцах`days_employed` и `total_income` имеются пропуска.

In [9]:
work_days = data.isnull()['days_employed'].sum()/len(data)
print('{:.2%}'.format(work_days))
pay = data.isnull()['total_income'].sum()/len(data)
print('{:.2%}'.format(pay))

10.10%
10.10%


Долю пропущенных значений в каждом из столбцов = **10.10%**

Причиной возникновению пропусков - человеческий фактор при заполнение/дубликате/переносе данных 

In [10]:
data['total_income'] = data['total_income'].fillna(data['total_income'].median())

Мы заменили все пустые значение в столбце `total_income` на медианное значение. Взятие медианное значение будет оптимальным варинатом закрыть пропуски в таблице, так как это будет более обьективным, чем среднее значение.

### Шаг 2.2 Проверка данных на аномалии и исправления.

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Например, отрицательное количество дней трудового стажа в столбце days_employed. Для реальных данных это нормально. Обработайте значения в столбцах с аномалиями и опишите возможные причины появления таких данных. После обработки аномалий заполните пропуски в days_employed медианными значениями по этому столбцу.

In [11]:
data['days_employed'] = data['days_employed'].abs()

Взяли столбец `days_employed` по модулю - abs() и таким образо избавились от знака **-**

По моему мнению, причина возникновения знака минус образовалось при транспортирвки цифр с одного файла в другой

In [12]:
data['days_employed'] = data['days_employed'].fillna(data['days_employed'].median())

Заменили все пустые значения в строках по столбцу `total_income` на медианной значение этого же столбца.

Все вышедшие строки с кол-во детей равным -1, заменим на 0.

In [13]:
data['children'] = data['children'].replace(-1, 0)

In [14]:
data.describe()

children  days_employed     dob_years  education_id  \
count  21525.000000   21525.000000  21525.000000  21525.000000   
mean       0.541092   60378.032733     43.293380      0.817236   
std        1.379943  133257.558514     12.574584      0.548138   
min        0.000000      24.141633      0.000000      0.000000   
25%        0.000000    1025.608174     33.000000      1.000000   
50%        0.000000    2194.220567     42.000000      1.000000   
75%        1.000000    4779.587738     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  2.152500e+04  
mean           0.972544      0.080883  1.651595e+05  
std            1.420324      0.272661  9.786607e+04  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.077982e+05  
50%            0.000000      0.000000  1.450179e+05  
75%            1.000000      0.000000  1.955436e+05  
max            4.000000      1.000000  2.265604e+06

### Шаг 2.3. Изменение типов данных.

Замените вещественный тип данных в столбце total_income на целочисленный, например, с помощью метода astype().

In [15]:
data['total_income'] = data['total_income'].astype('int')

In [16]:
data['days_employed'] = data['days_employed'].astype('int')

C помощью метода **astype()** меняем тип данных столбца `total_income` и `days_employed` на целые числа. Теперь все данные представлены в удобном формате.

### Шаг 2.4. Удаление дубликатов.

Если в данных присутствуют строки-дубликаты, удалите их. Также обработайте неявные дубликаты. Например, в столбце education есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к одному регистру. Проверьте остальные столбцы.

После удаления дубликатов сделайте следующее:

поясните, как выбирали метод для поиска и удаления дубликатов в данных;

приведите возможные причины появления дубликатов.

Использую метод **duplicated() и sum()**, выводим сумарное кол-во дубликатов.

In [17]:
data.duplicated().sum()

54

Теперь выведем таблицу с дубликатами, что б наглядно на них посмотреть

In [18]:
data[data.duplicated()].sort_values(by = ['days_employed', 'total_income']).head()

children  days_employed  dob_years education  education_id  \
2849         0           2194         41   среднее             1   
4182         1           2194         34    ВЫСШЕЕ             0   
4851         0           2194         60   среднее             1   
5557         0           2194         58   среднее             1   
7808         0           2194         57   среднее             1   

         family_status  family_status_id gender income_type  debt  \
2849   женат / замужем                 0      F   сотрудник     0   
4182  гражданский брак                 1      F   сотрудник     0   
4851  гражданский брак                 1      F   пенсионер     0   
5557  гражданский брак                 1      F   пенсионер     0   
7808  гражданский брак                 1      F   пенсионер     0   

      total_income                  purpose  
2849        145017  покупка жилья для семьи  
4182        145017                  свадьба  
4851        145017                  свадьба  
5557        145017          сыграть свадьбу  
7808        145017    на проведение свадьбы

Да, по таблице видны дубликаты. **Причина?** Возможно задвоение данных. 

Теперь их нужно стереть с наших глаз, а то мазолят)))

In [19]:
data = data.drop_duplicates()
data.duplicated().sum()

0

Удалили и проверили, все чисто. Теперь переходим к обработке к неявным дубликатам, найдем их с помощью метода **unique()**. Начнем со столбца `education`

In [20]:
data['education'].unique()

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

Мы видим кашу малашу в столбце `education`, а именно верхние регистры, заглвные буквы. Для человека самостоятельно изучающий DS было бы нормой, но для меня нет, поэтому использую код **.str.lower()** для столбца `education`, чтобы все слова были нижнего регистра и далее вывожу таблицу и смотрю сработал код или нет.

In [21]:
data['education'] = data['education'].str.lower()

In [22]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

В точку, все слова в таблице `education` поменяли на нижний регистр.

### Шаг 2.5. Формирование дополнительных датафреймов словарей, декомпозиция исходного датафрейма.

Создайте два новых датафрейма, в которых:

каждому уникальному значению из education соответствует уникальное значение education_id — в первом;

каждому уникальному значению из family_status соответствует уникальное значение family_status_id — во втором.

Удалите из исходного датафрейма столбцы education и family_status, оставив только их идентификаторы: education_id и family_status_id. Новые датафреймы — это те самые «словари» (не путайте с одноимённой структурой данных в Python), к которым вы сможете обращаться по идентификатору.

**НЕ ПОНЯЛ КАК ГРАМОТНО СДЕЛАТЬ В ЭТОМ ШАГЕ**

Создаем два новых датафрейма `edu`, `fam`. После этого используем функцию **drop()** для удаления указанных столбцов, **inplcae=True** создает нам новое изменение в новом обькте, который создается, а **axis=1** предназачен для операция по столбцам

In [23]:
edu = data[['education', 'education_id']]
fam = data[['family_status', 'family_status_id']]

In [24]:
edu.drop_duplicates()
fam.drop_duplicates()
data.head()

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

In [26]:
data.drop('education', inplace=True, axis=1)
data.drop('family_status', inplace=True, axis=1)
data.head()

children  days_employed  dob_years  education_id  family_status_id gender  \
0         1           8437         42             0                 0      F   
1         1           4024         36             1                 0      F   
2         0           5623         33             1                 0      M   
3         3           4124         32             1                 0      M   
4         0         340266         53             1                 1      F   

  income_type  debt  total_income                     purpose  
0   сотрудник     0        253875               покупка жилья  
1   сотрудник     0        112080     приобретение автомобиля  
2   сотрудник     0        145885               покупка жилья  
3   сотрудник     0        267628  дополнительное образование  
4   пенсионер     0        158616             сыграть свадьбу

**Категоризация:**

Стаж клиентов `days_employed`:
- до 15 лет
- 15-30 лет
- более 30 лет

Возраст клиентов `dob_years`:
- до 25 лет
- 25-45 лет
- 45-65 лет
- более 65 лет

Кол-во детей `children`:
- 0 детей
- 1-2 ребенок/ребенка
- 2+ ребека

И добавим все новые столбцы в таблицу `data`

In [27]:
def days_employed_type(row):
    days_employed = row['days_employed']
    
    if days_employed <= 5400:
        return 'Стаж до 15 лет'
    if 5401 <= days_employed <= 10800:
        return 'Стаж от 15 до 30 лет'
    else:
        return 'Стаж более 30 лет'
    
def dob_years_type(row):
    dob_years = row['dob_years']
    
    if dob_years < 25:
        return 'До 25 лет'
    if 25 <= dob_years < 45:
        return 'От 25 до 25 лет'
    if 45 <= dob_years < 65:
        return 'От 45 до 65 лет'
    else:
        return 'Более 65 лет'

def children_type(row):
    children = row['children']
    
    if children == 0:
        return 'Нет детей'
    if 1 <= children <= 2:
        return '1-2 ребенка'
    else:
        return '3+ ребенка'

In [28]:
data['days_employed_type'] = data.apply(days_employed_type, axis=1)
data['dob_years_type'] = data.apply(dob_years_type, axis=1)
data['children_type'] = data.apply(children_type, axis=1)

### Шаг 2.6. Категоризация дохода.

На основании диапазонов, указанных ниже, создайте столбец total_income_category с категориями:

0–30000 — 'E';

30001–50000 — 'D';

50001–200000 — 'C';

200001–1000000 — 'B';

1000001 и выше — 'A'.

Например, кредитополучателю с доходом 25000 нужно назначить категорию 'E', а клиенту, получающему 235000, — 'B'.

In [29]:
def total_income_category(row):
    total_income = row['total_income']
    
    if 0 <= total_income <= 30000:
        return 'E'
    if 30001 <= total_income <= 50000:
        return 'D'
    if 50001 <= total_income <= 200000:
        return 'C'
    if 200001 <= total_income <= 1000000:
        return 'B'
    if 1000001 <= total_income :
        return 'A'
    
row_values = [25000]
row_columns = ['total_income']
row = pd.Series(data=row_values, index=row_columns)
data['total_income_category'] = data.apply(total_income_category, axis=1)
total_income_category(row)

'E'

In [30]:
data['total_income_category'].value_counts()

C    16033
B     5041
D      350
A       25
E       22
Name: total_income_category, dtype: int64

Сделали категоризацию по столбцу `total_income` и для каждой критерии свои условия, на вход у нас попадает ежемесечная заработная плата ( или просто рандомное число ) а возращает его/ее категорию. По примеру категоризация работает правильно. Осталось создать отдельный столбец и для этого используем функцию **apply()**.

Дополнительно вывел статистику по доходности, и теперь видно сколько ориентировочно зарабатывает основная масса.

### Шаг 2.7. Категоризация целей кредита.

Создайте функцию, которая на основании данных из столбца purpose сформирует новый столбец purpose_category, в который войдут следующие категории:

'операции с автомобилем',

'операции с недвижимостью',

'проведение свадьбы',

'получение образования'.

Например, если в столбце purpose находится подстрока 'на покупку автомобиля', то в столбце purpose_category должна появиться строка 'операции с автомобилем'.

Вы можете использовать собственную функцию и метод apply(). Изучите данные в столбце purpose и определите, какие подстроки помогут вам правильно определить категорию.

In [31]:
def purpose_category(row):
    purpose = row['purpose']
    
    if 'авто' in purpose:
        return 'операции с автомобилем'
    if 'недвиж' in purpose or 'жиль' in purpose:
        return 'операции с недвижимостью'
    if 'свадьб' in purpose:
        return 'проведение свадьбы'
    if 'образован' in purpose:
        return 'получение образования'
    
row_values = ['на покупку автомобиля']
row_columns = ['purpose']
row=pd.Series(data=row_values, index=row_columns)
data['purpose_category'] = data.apply(purpose_category, axis=1)
purpose_category(row)

'операции с автомобилем'

In [32]:
data['purpose_category'].value_counts()

операции с недвижимостью    10814
операции с автомобилем       4308
получение образования        4014
проведение свадьбы           2335
Name: purpose_category, dtype: int64

### Ответы на вопросы.

Пишим функцию для вывода информации.

In [33]:
children_group = data.groupby('children_type')['debt'].agg(['count', 'sum'])
children_group['conversion'] = (children_group['sum'] / children_group['count']) * 100
children_group.sort_values('conversion')

count   sum  conversion
children_type                         
Нет детей      14154  1064    7.517310
3+ ребенка       456    39    8.552632
1-2 ребенка     6861   638    9.298936

##### Вопрос 1:

Есть ли зависимость между количеством детей и возвратом кредита в срок?



##### Вывод 1:
Да, есть зависимость, клиенты у которых нет детей имеют низжие процент задолжности.

In [34]:
family_status_id_group = data.groupby('family_status_id')['debt'].agg(['count', 'sum'])
family_status_id_group['conversion'] = (family_status_id_group['sum'] / family_status_id_group['count']) * 100
family_status_id_group.sort_values('conversion')

count  sum  conversion
family_status_id                        
2                   959   63    6.569343
3                  1195   85    7.112971
0                 12344  931    7.542126
1                  4163  388    9.320202
4                  2810  274    9.750890

##### Вопрос 2:

Есть ли зависимость между семейным положением и возвратом кредита в срок?



##### Вывод 2:
Первым кто имеет не большой процент задолжности кредита это вдовы, на удивление люди в официальном браке остаются в небольшом проценте чем люди в разводе. Так что да, зависимость есть.

In [35]:
total_income_category_group = data.groupby('total_income_category')['debt'].agg(['count', 'sum'])
total_income_category_group['conversion'] = (total_income_category_group['sum'] / total_income_category_group['count']) * 100
total_income_category_group.sort_values('conversion')

count   sum  conversion
total_income_category                         
D                        350    21    6.000000
B                       5041   356    7.062091
A                         25     2    8.000000
C                      16033  1360    8.482505
E                         22     2    9.090909

In [ ]:
info = data.merge(fam, on='family_status_id')
data_temp = info.pivot_table(index='family_status', values='debt', aggfunc=['count', 'sum', 'mean'])
data_temp.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля должников']
display(data_temp)

##### Вопрос 3:

Есть ли зависимость между уровнем дохода и возвратом кредита в срок?



##### Вывод 3:
На удивление, люди миллионеры имеют % задожности больше чем у людей зарпалата от 30К до 50К. 

In [ ]:
purpose_category_group = data.groupby('purpose_category')['debt'].agg(['count', 'sum'])
purpose_category_group['conversion'] = (purpose_category_group['sum'] / purpose_category_group['count']) * 100
purpose_category_group.sort_values('conversion')

##### Вопрос 4:

Как разные цели кредита влияют на его возврат в срок?



##### Вывод 4:
Это правильно, что люди с целью приобрести недвижимость имеют наименьший процент по задолжности, это отличный вариант. А вот люди берущие для образование и авто нарушают выплоты кредита.

## Общий вывод:

Общий вывод таков:

Надежными клиентами которые показали низкий процент по задолжностям являются люди не имеющих детей, которые ставят цель приобрести недвижимость, имеют среднюю заработную плату по России и отдельно люди вдовы.

Не надежными клиентами являются с целью приобрести авто, с з/п до 30К и который состоят в простых отношениях, но при этом могут иметь по 1-2 ребенка.